src/client/client_impl.cc

```diff
@@ -584,21 +602,15 @@ ClientImpl::ClientImpl(common::ConfigManager* config_manager,
   cache_manager_ = CacheManager::ConstructCacheManager(
       config_manager_->GetClientCacheTimeout());
 
+  resolve_hostname_ = resolve_hostname;
```

Comment:
nit. this line should be moved to the constructor of ClientImpl, which already takes in the resolve_hostname.